packages/core/src/utils/transformers/isFormatSupported.ts

```diff
@@ -0,0 +1,44 @@
+const supportedFormats: {
+  webp?: { checked: boolean; supported?: boolean };
+  avif: { checked: boolean; supported?: boolean };
+} = {
+  webp: { checked: false },
+  avif: { checked: false },
+};
+
+const methodMap = {
+  webp: supportsWebp,
+  avif: supportsAvif,
+};
+
+export async function isImageFormatSupported(format: 'webp' | 'avif') {
+  if (supportedFormats[format]?.checked) {
+    return supportedFormats[format]?.supported;
+  } else {
+    const supported = await methodMap[format]();
+    supportedFormats[format] = { checked: true, supported };
+    return supported;
+  }
+}
+
+async function supportsWebp() {
+  if (typeof window === 'undefined') return false;
+  const canvas = document.createElement('canvas');
+  if (!canvas.toDataURL) return false;
+  const imageData = canvas.toDataURL('image/webp');
+  if (imageData.startsWith('data:image/webp')) {
+    return true;
+  }
+}
+
+//Most browsers don't support avif exporting in canvas yet, so have to use another method for avif
+async function supportsAvif() {
+  if (typeof window === 'undefined' || !window.createImageBitmap) return false;
```

Comment:
This means if we do SSR, we only support these once hydration kicks in :thinking:

Another way could be [progressive enhancement based](https://darekkay.com/blog/avif-images/#progressive-enhancement) on the `picture` tag. But we would have to account for it in the `OptimizedImageAsset` then.

Reply:
ya its a good point, and I don't think this will be fully implemented for the GA release. I had to put it on the back burner.